### PR TITLE
ci: Add docker credentials

### PR DESCRIPTION
**Pull request by mikolajb**
Let's fix the following problem:

```
(docker image inspect mesosphere/mesosphere/dkp-catalog-applications-ci:73fbedfa0a22d88a7b64f7c5d728706a8c7e94d4 &>/dev/null && echo 'mesosphere/mesosphere/dkp-catalog-applications-ci:73fbedfa0a22d88a7b64f7c5d728706a8c7e94d4 already exists - skipping image build' ) || \
    docker pull mesosphere/mesosphere/dkp-catalog-applications-ci:73fbedfa0a22d88a7b64f7c5d728706a8c7e94d4 || \
    make ci.docker.build
Error response from daemon: pull access denied for mesosphere/mesosphere/dkp-catalog-applications-ci, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

https://github.com/mesosphere/dkp-catalog-applications/actions/runs/6242181447/job/16958359076#step:7:18

Also, changing the value of `GITHUB_REPOSITORY` as it is a variable provided by the GHA env: https://docs.github.com/en/actions/learn-github-actions/variables

### CODE REVIEW

**Review by tillt — APPROVED** [review]

File: .github/workflows/release-airgap-artifacts.yaml [hunk 1]
```diff
@@ -13,6 +13,11 @@ jobs:
     runs-on:
       - self-hosted
       - medium
+
+    env:
```
Comment (cbuto) [inline .github/workflows/release-airgap-artifacts.yaml:17]: it looks like we don’t have these explicitly set in the other release branches. should we backport the full PR and see if that works? 

i.e. https://github.com/mesosphere/dkp-catalog-applications/pull/107

Comment (mikolajb) [reply .github/workflows/release-airgap-artifacts.yaml:17]: Absolutely! 💯

**Comment by mikolajb** [comment]
> Shouldn't this use nexus and appropriate credentials?

Thanks Jimmi! @shaeed-noordeen I switch to nexus credentials, is it the right thing to do?

**Review by Azhovan — APPROVED** [review]

**Comment by shaeed-noordeen** [comment]
Think the issue lies with the docker image name variable -
It should be mesosphere/dkp-catalog-applications-ci

There is an extra “mesosphere/” in the error message

**Comment by shaeed-noordeen** [comment]
> > Shouldn't this use nexus and appropriate credentials?
> 
> Thanks Jimmi! @shaeed-noordeen I switch to nexus credentials, is it the right thing to do?

Yeah we can do that

**Comment by mikolajb** [comment]
> Think the issue lies with the docker image name variable - It should be mesosphere/dkp-catalog-applications-ci
> 
> There is an extra “mesosphere/” in the error message

Good catch! It does seem like a case! Thank you!
https://docs.github.com/en/actions/learn-github-actions/variables

**Pull request merged**